hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/upgrade/ReconLayoutVersionManager.java

```diff
@@ -47,12 +48,12 @@ public class ReconLayoutVersionManager {
   private int currentMLV;
 
   public ReconLayoutVersionManager(ReconSchemaVersionTableManager schemaVersionTableManager,
-                                   ReconContext reconContext)
+                                   ReconContext reconContext, Injector injector)
```

Comment:
You need not to pass Injector reference here, as injector is only used to get instance once which you have binded in guice framework as Singleton. And those all `ReconUpgradeAction` classes will not be used in code flow anywhere. So impl classes of `ReconUpgradeAction` will be scanned using annotation and you can create instance of it there itself in your `registerUpgradeActions` scan code.